src/util.cpp

```diff
@@ -579,6 +579,26 @@ const fs::path &GetDataDir(bool fNetSpecific)
     return path;
 }
 
+fs::path GetWalletDir()
+{
+    fs::path path;
+
+    if (gArgs.IsArgSet("-walletdir")) {
+        path = fs::system_complete(gArgs.GetArg("-walletdir", ""));
+        if (!fs::is_directory(path)) {
+            path = "";
```

Comment:
At the moment `GetWalletDir` is used in the startup and in runtime.

IMO, in runtime, this should not be possible. Create a `bool IsValidWalletDir()` (or whatever) to use in startup and here assert `is_directory`?

Reply:
Yep I'll include something similar when reworking the directory caching stuff I think, datadir and walletdir shouldn't change throughout execution so no point rechecking every time